internal/tokens/urltokens.go

```diff
@@ -0,0 +1,220 @@
+package tokens
+
+import (
+	"crypto/hmac"
+	"crypto/rand"
+	"crypto/sha256"
+	"encoding/base64"
+	"time"
+
+	"github.com/oklog/ulid/v2"
+	"github.com/vmihailenco/msgpack/v5"
+
+	ulids "github.com/datumforge/datum/internal/utils/ulids"
+)
+
+const (
+	nonceLength                 = 64
+	keyLength                   = 64
+	expirationDays              = 7
+	resetTokenExpirationMinutes = 30
+)
+
+// NewVerificationToken creates a token struct from an email address that expires
+// in 7 days
+func NewVerificationToken(email string) (token *VerificationToken, err error) {
+	if email == "" {
+		return nil, ErrMissingEmail
+	}
+
+	token = &VerificationToken{
+		Email: email,
+	}
+
+	if token.SigningInfo, err = NewSigningInfo(time.Hour * 24 * expirationDays); err != nil {
+		return nil, err
+	}
+
+	return token, nil
+}
+
+// VerificationToken packages an email address with random data and an expiration
+// time so that it can be serialized and hashed into a token which can be sent to users
+type VerificationToken struct {
+	Email string `msgpack:"email"`
+	SigningInfo
+}
+
+// Sign creates a base64 encoded string from the token data so that it can be sent to
+// users as part of a URL. The returned secret should be stored in the database so that
+// the string can be recomputed when verifying a user provided token.
+func (t *VerificationToken) Sign() (_ string, secret []byte, err error) {
+	var data []byte
+
+	if data, err = msgpack.Marshal(t); err != nil {
+		return "", nil, err
+	}
+
+	return t.signData(data)
+}
```

Comment:
```suggestion
func (t *VerificationToken) Sign() (string, []byte, error) {
	var data []byte

	if data, err := msgpack.Marshal(t); err != nil {
		return "", nil, err
	}

	return t.signData(data)
}
```

you aren't using the name vars, so doesn't make sense to declare them